packages/owl-bot/src/handlers.ts

```diff
@@ -335,3 +342,95 @@ Please fix this as soon as possible so that your repository will not go stale.`;
     );
   }
 }
+
+/**
+ * Invoked when someone edits the title of body of a pull request.
+ * We're interested if they checked the box to "regenerate this pull request."
+ */
+export async function handlePullRequestEdited(
+  appId: number,
+  privateKey: string,
+  project: string,
+  trigger: string,
+  payload: PullRequestEditedEvent,
+  octokit: OctokitType,
+  logger = console
+) {
+  const head = payload.pull_request.head.repo.full_name;
+  const base = payload.pull_request.base.repo.full_name;
+  const [owner, repo] = base.split('/');
+  const installation = payload.installation?.id;
+  const prNumber = payload.pull_request.number;
+
+  if (!installation) {
+    throw Error(`no installation token found for ${head}`);
+  }
+
+  const token = await core.getGitHubShortLivedAccessToken(
+    privateKey,
+    appId,
+    installation
+  );
+
+  const oldBody = payload.pull_request.body ?? '';
+  const newBody = payload.changes.body?.from ?? '';
+
+  if (
+    oldBody.includes(REGENERATE_CHECKBOX_TEXT) ||
+    !newBody.includes(REGENERATE_CHECKBOX_TEXT)
+  ) {
+    logger.info(
+      "The user didn't check the regenerate me box for PR #{prNumber}"
+    );
+    return;
+  }
+
+  // No matter what the outcome, we'll create a comment below.
+  const createComment = (body: string): void => {
+    octokit.issues.createComment({
+      owner,
+      repo,
+      issue_number: prNumber,
+      body,
+    });
+  };
+
+  // The user checked the "Regenerate this pull request" box.
+
+  const sourceHash = findSourceHash(newBody);
+  if (!sourceHash) {
+    // But there's no source hash to regenerate from.  Oh no!
+    const sourceLine = sourceLinkLineFrom(sourceLinkFrom('abc123'));
+    createComment(`Owl Bot could not regenerate this pull request because the body is missing a source hash.
+
+A source hash in the source link looks like this:
+${sourceLine}`);
+    return;
+  }
+
+  try {
+    const cb = core.getCloudBuildInstance();
+    await cb.runBuildTrigger({
```

Comment:
I believe this will return a long running operation which we then need to await completion on, this is what's happening from line `134` on here:

https://github.com/googleapis/repo-automation-bots/blob/ae6615f4dfc002bdd105f58e79f85927d8834c1c/packages/owl-bot/src/core.ts#L134

What we could perhaps do is pull this into a `cloud-build.ts` wrapper, that handles this nuisance for us.

Reply:
Ok.  Will do.

Reply:
I think I got confused because I'd expect functions named `handlePullRequestLabeled()` and `handlePullRequestEdited()` to live in a file called `handlers.ts`.

Reply:
In this case, there is no github check that we need to update.  The build job does the work of updating the pull request with either a success or failure.  What would we do after we waited for the result here?